### PR TITLE
Don't add @princeton.edu to UIDs when they are email addresses already

### LINKED ISSUE #1482: Email addresses are wrong for non-Princeton users
It's possible for non-Princeton users to get accounts, and their NetIDs are their (external) email addresses.  We currently just add `@princeton.edu` to make their email addresses, resulting in malformed email addresses like `foo@bar.com@princeton.edu`

### PR DESCRIPTION
**Pull request by escowles**
Fixes #1482

### CODE REVIEW

**Comment by tpendragon** [comment]
Should we have a migration for existing users?

**Comment by escowles** [comment]
@tpendragon Yep, just added that as a rake task.

**Comment by tpendragon** [comment]
@escowles Can you rebase this? I expect it to fail CI now, since this should be in release notes.

**Comment by escowles** [comment]
👍 I rebased and added a release notes entry (also opened https://github.com/pulibrary/figgy/pull/3248 with a straight rebase to make sure it failed, and it does)

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ Release notes template:
 
 ## Fixed
 
+* Email addresses for external users no longer have "@princeton.edu" added to the end of valid email addresses.
```
Comment (tpendragon) [inline RELEASE_NOTES.md]: I think you added this to the documentation template instead of as a new entry.

Comment (escowles) [reply RELEASE_NOTES.md]: 😞 yes, that's exactly what I did.

**Pull request merged**